writer.go

```diff
@@ -2,20 +2,189 @@ package warc
 
 import (
 	"bytes"
+	"compress/gzip"
+	stdErrors "errors"
 	"fmt"
 	"io"
 	"net/http"
 	"sort"
 
 	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
 )
 
 // NewUUID generates a new version 4 uuid
 func NewUUID() string {
 	return fmt.Sprintf("<urn:uuid:%s>", uuid.New())
 }
 
+type flusher interface {
+	Flush() error
+}
+
+// Writer provides functionality for writing WARC files in compressed and
+// uncompressed formats.
+//
+// To construct a Writer, call NewWriterCompressed or NewWriterRaw.
+type Writer struct {
+	seekW io.WriteSeeker
+	wr    io.Writer
+
+	// RecordCallback will be called after each record is written to the file.
+	// If a WriteSeeker was not provided, the provided positions will be
+	// invalid.
+	RecordCallback func(r *Record, startPos, endPos int64)
```

Comment:
Hmm, I'll show you the client code I have for this - I'm using the callback to write an old-style CDX, and using a channel would just have me attempting to receive after writing each record instead.

Now that you point out having a WrittenRecord type, it might be better to just make that be the return value of the function?